### DISCUSSION TITLE
Kata10

### DISCUSSION THREAD

**Discussion by AndresAVargas** [LaunchX-InnovaccionVirtual/CursoIntroPython#276] [Q&A]
Hola compañero ya me salio el primer error al ejecutar el open.py pero al tratar de ejecutar los comandos de abajo me aparece esto, creo que lo estoy escribiendo mal o nose ayuda :CCC
![image](https://user-images.githubusercontent.com/88987587/154172989-6d65c2a0-2530-49cb-96c3-ed8a9178fec2.png)

**Comment by doguedogue** [comment]
En el que gustes, ya te falta poco animo, checa otras preguntas, sobre esta kata se ha hablado bastante.
🚀🚀🚀

**Reply by AndresAVargas** [reply]
es que estoy confundido, entiendo que hay que hacer 2 archivos .py y luego ejecutarlos en el cmd, y segun yo tienes que resolver ese problema pero lo que no entiendo es como escribir eso que viene escrito si es copiar y pegar y tenemos que escribir los comandos mas cortos

**Comment by OscarPedroza** [comment]
Introduce solo el código sin los símbolos de mayor que y puntos de la izquierda '>> ...'.

**Reply by AndresAVargas** [reply]
eso intento de echo logre pasar a donde creamos un directorio con nombre config.txt, pero no se como actualizar la funcion, no se si borrar la primera que ya se habia echo o que hacer me trabe aqui
![image](https://user-images.githubusercontent.com/88987587/154176459-35caa2ea-7956-4ca3-abb0-750994affcd7.png)
segun tiene que salir el mensaje de print y no me da, aun borrando la funcion de arriba no me da el mensaje 
![image](https://user-images.githubusercontent.com/88987587/154176550-e715ab22-46d4-49b5-b61c-d58276634519.png)
al contrario, si pasa el comando

**Reply by irvingfloley** [reply]
Te voy a comentar por que me ocurrió lo mismo, al seguir las instrucciones pareciera que te da a entender que debes cambiar por completo el archivo "config.py", pero en tu caso en la segunda captura veo que borraste este código:

`if __name__ == '__main__':
    main()`

Sin eso no veras el print ya que tu main no se ejecutara, error de principiantes en el lenguaje, la verdad creo que el modulo 10 fue el más difícil de seguir de todos, también me encontré con unas diferencias ya que al parecer las pruebas que usaron en el modulo las hicieron en Linux y en Windows varían un poco las excepciones arrojadas, o no se si fue la versión de Python, pero no pude replicar el error de tipo IsDirectory.

**Reply by AndreCerva** [reply]
Coincido con @irvingfloley, mira, en la primera captura llamas a la función **main** pero hay dos funciones con ese nombre, eso confunde a tu código, lo que deberías hacer es llamarlas distinto a cada una y llamar a cada una independientemente, para evitarte esto, podrías hacerlo en un .ipynb donde no necesitarías crear una función y llamarla, bastaría con ejecutar bloque por bloque.

En tu segunda captura, lo que pasa es que la instrucción que borraste:
 `if __name__ == '__main__':
       main()`
hace que nunca vaya hacia tu función main, ya que en tu código nunca la estás llamando, te explico, cuando python ejecuta un script, crea algunas variables especiales, en las que dentro de esas está  `__name__`, ahora, cuando este script se ejecuta directamente (entiéndase directamente como correr un programa como lo hemos estado haciendo hasta ahora, desde la terminal o dando click en correr con alguna extensión en vs code o en el botón de run de Jupyter notebook) la variable especial `__name__` toma el valor de: `'__main__'`, por ello el condicional **if**, cuando se ejecuta el script como el programa principal(ejecutarse directamente) se cumple la condición y entra a lo que hay dentro, donde se encuentra **main()** esto lo que hace es llamar a la función que está declarada arriba de estas dos líneas de código y tu borraste este proceso, por lo que nunca llamas a tu función, por lo que nunca se ejecutará lo que hay dentro de esta. La solución a esto es simplemente volver a poner el condicional  `if __name__ == '__main__': main()`

También lo que comenta irving es cierto, si no te sale el mismo Traceback que en la documentación de Fer, muy probablemente se deba a la versión de Python, de igual forma, no te preocupes por ello, lo importante es entender la importancia de manejar excepciones, tu puedes =)
